inc/os_linux.h

```diff
@@ -9,6 +9,7 @@
 //
 #define _GNU_SOURCE
 #define __USE_GNU
+#define __USE_MISC // needed for SOL_TCP in <netinet/tcp.h>
```

Comment:
Yep, same here.  But the biggest problem is that os.h or concrete os_posix.h etc. are included all over the place int the pal, and after some OS headers such as string.h or stdio, etc. are already included, thus feature test macros are already evaluated and these settings do not grab hold.  What I did and what I am testing now is to a) move all os.h to the beginning of the .c, then b) set 
```
//
// Linux OS api used in PAL implementation
//
#define _DEFAULT_SOURCE 1
#define _BSD_SOURCE 1
#define _SVID_SOURCE 1
#define _XOPEN_SOURCE 600
#define _GNU_SOURCE 1
#define _FORTIFY_SOURCE 1
#define _ISOC99_SOURCE 1
#include <features.h>

#include "os_posix.h"
```
I am building this now on all distros and let you know what I find.